src/Core/Product/Combination/NameBuilder/CombinationNameBuilder.php

```diff
@@ -28,23 +28,68 @@
 namespace PrestaShop\PrestaShop\Core\Product\Combination\NameBuilder;
 
 use PrestaShop\PrestaShop\Core\Domain\Product\Combination\CombinationAttributeInformation;
+use Symfony\Component\Translation\TranslatorInterface;
 
 /**
  * Builds combination name by attributes information
  */
 class CombinationNameBuilder implements CombinationNameBuilderInterface
 {
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+    /**
+     * @var string
+     */
+    private $attributesSeparator;
+
+    public function __construct(
+        TranslatorInterface $translator,
+        string $attributesSeparator
+    ) {
+        $this->translator = $translator;
+        $this->attributesSeparator = $attributesSeparator;
+    }
+
     /**
      * {@inheritdoc}
      */
     public function buildName(array $attributesInfo): string
     {
-        return implode(', ', array_map(function (CombinationAttributeInformation $combinationAttributeInfo): string {
-            return sprintf(
-                '%s - %s',
-                $combinationAttributeInfo->getAttributeGroupName(),
-                $combinationAttributeInfo->getAttributeName()
-            );
-        }, $attributesInfo));
+        return implode(
+            $this->attributesSeparator,
+            array_map(
+                ['$this', 'translateAttribute'],
```

Comment:
```suggestion
                [$this, 'translateAttribute'],
```

Reply:
I believe this is the correct syntax

Reply:
Blocked with other issue I cannot perform any test.... but it seems better